### PR TITLE
Playground js molecule field

### PR DESCRIPTION
**Pull request by juanmaguitar**
Use of new `demo` folder to allow import of external dependencies
Use cases of `MoleculeField` component integrated with `FormInput` & `AtomTextarea`
Also use case of `MoleculeField` inside of a dynamic form with validation rules

### CODE REVIEW

File: demo/molecule/field/demo/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@s-ui/react-atom-textarea": "^1.2.0",
```
Comment (davidbarna) [inline demo/molecule/field/demo/package.json]: use major/exact versions

File: demo/molecule/field/demo/FormWithFormik/index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/* eslint-disable react/prop-types, no-unused-vars */
+
+import React from 'react'
+import {withFormik, Field} from 'formik'
```
Comment (davidbarna) [inline demo/molecule/field/demo/FormWithFormik/index.js]: Were are you using `Field` ?

File: demo/molecule/field/demo/MoleculeFieldTextarea/hoc/WithCharacterCount.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import React, {Component} from 'react'
+
+import AtomTextarea from '@s-ui/react-atom-textarea'
+
+const getHelpTextArea = (numCharacters, maxCharacters) => {
+  return `${numCharacters}/${maxCharacters} characters`
+}
+
+const WithCharacterCount = BaseComponent => {
```
Comment (davidbarna) [inline demo/molecule/field/demo/MoleculeFieldTextarea/hoc/WithCharacterCount.js]: Isn't this should be as a service of the field instead of demo ?

**Review by davidbarna — DISMISSED** [review]

**Comment by jcger** [comment]
Enablers will discuss about if it's ok to use formik

File: demo/molecule/field/demo/index.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
+@import '~@s-ui/react-atom-textarea/lib/index';
```
Comment (jcger) [inline demo/molecule/field/demo/index.scss]: you need to add the input css too

File: demo/molecule/field/demo/package.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@s-ui/react-atom-spinner": "1",
+    "@s-ui/react-atom-textarea": "1",
+    "@s-ui/react-form-input": "1",
+    "@s-ui/react-molecule-field-textarea": "1",
+    "formik": "^1.0.2",
```
Comment (jcger) [inline demo/molecule/field/demo/package.json]: use exact version please, you can npm install with -DE (save dev exact) or -SE (save exact) in order do not add the ^

File: demo/molecule/field/demo/index.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react/prop-types, no-unused-vars, no-console */
+
+import React from 'react'
+
+import MoleculeField from '../../../../components/molecule/field/src'
+
+import MoleculeFieldTextarea from '@s-ui/react-molecule-field-textarea'
```
Comment (jcger) [inline demo/molecule/field/demo/index.js]: you are not using this, right?

File: demo/molecule/field/demo/FormWithFormik/index.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+            }
+            placeholder="Tell us about yourself"
+            onChange={({value}) => setFieldValue('description', value)}
+            value={values.description}
+            name="description"
+            maxCharacters={30}
+          />
+        </div>
+      </div>
+      <button disabled={isSubmitting} type="submit">
```
Comment (jcger) [inline demo/molecule/field/demo/FormWithFormik/index.js]: you could use the @schibstedspain/sui-atom-button

File: demo/molecule/field/demo/index.scss (hunk 4 above)
Comment (juanmaguitar) [reply demo/molecule/field/demo/index.scss]: Added!

**Review by jcger — DISMISSED** [review]

**Review by davidbarna — DISMISSED** [review]

**Comment by juanmaguitar** [comment]
Going to wait for the approval of the `MoleculeFieldInput` (https://github.com/SUI-Components/sui-components/pull/507) and its integration in this demo, to move on this PR

**Comment by juanmaguitar** [comment]
Waiting for #516 to link proper `MoleculeTextareaField` in this demo

**Review by jcger — DISMISSED** [review]

**Review by jcger — DISMISSED** [review]

**Review by jcger — APPROVED** [review]

**Review by carlosvillu — APPROVED** [review]

**Pull request merged**